src/task.rs

```diff
@@ -70,10 +70,18 @@ fn run_task(
     cmdline: &str,
     depfile: Option<&str>,
     rspfile: Option<&RspFile>,
+    dry_run: bool,
 ) -> anyhow::Result<TaskResult> {
     if let Some(rspfile) = rspfile {
         write_rspfile(rspfile)?;
     }
+    if dry_run {
```

Comment:
Hm, is it intentional here to write the rspfile in the dry run case?  I am not sure about it, but my first impression is we shouldn't write any files, and perhaps even not start any tasks.

Reply:
Ninja does write rsp files in dry run mode. I know that some people use `-n` to generate rsp files.

Then again, n2 doesn't delete rsp files by default, so maybe that's less needed here.

It *is* weird if `-n` mode has visible side effects, and using it for writing rsp files feels like a bit of a hack.

Reply:
What's the use case for generating rsp files with -n?  Debugging?

Reply:
My intent here was to match ninja - but upon reflection I agree that the ninja behavior is strange. Is the goal of n2 in general to match behaviors or to rethink them? I.e. do you anticipate having a `keeprsp` control as well?

I've never intentionally used the `-n` flag to examine rspfiles but I've certainly used and seen folks deeply confused by ninja's keeprsp behehavior. Perhaps there's a better way to approach that entirely.

Reply:
> What's the use case for generating rsp files with -n? Debugging?

I don't use it for this myself, so I'm not 100% sure, but I think the answer is "yes". These are concrete examples:

1. Being able to manually run commands. You might want to run with `-v` to get a commandline, but if that uses a rsp file and it's not on disk and you forgot to pass `-d keeprsp`, you can re-run with `-n` to create the rsp file.
2. You just might want to read the rsp file to look at its contents.